src/pages/ConnectWallet/MobileConnect.tsx

```diff
@@ -123,22 +121,10 @@ export const MobileConnect = () => {
           if (!(await wallet?.isInitialized())) {
             await wallet?.initialize()
           }
-          dispatch({
-            type: WalletActions.SET_WALLET,
-            payload: {
-              wallet,
-              name,
-              icon,
-              deviceId,
-              meta: { label: item.label },
-              connectedType: KeyManager.Mobile,
-            },
-          })
-          dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
-          dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
```

Comment:
@NeOMakinG let's comment on this diff (in the PR), I read this a few times yesterday and today and still can't decipher it without diving into the implementation.

I suppose the intent here is that `load` *should* dispatch all these?

Reply:
Not valid anymore as discussed we prefer the quickfix regarding the complexity of `WalletProvider`, changed in [5eacb01](https://github.com/shapeshift/web/pull/7280/commits/5eacb015dbea0fa87977f66ebe16bb50b6338894)